src/components/frontend-engine/frontend-engine.tsx

```diff
@@ -63,18 +73,42 @@ const FrontendEngineInner = forwardRef<IFrontendEngineRef, IFrontendEngineProps>
 		onSubmit?.(data);
 	};
 
+	const setErrors = (errors: Record<string, string | string[]>): void => {
+		for (const key in errors) {
+			if (key in fields) {
+				const errorMessage = errors[key];
+
+				if (Array.isArray(errorMessage)) {
+					setError(key, { type: "api", message: errorMessage[0] });
+				} else {
+					setError(key, { type: "api", message: errorMessage });
+				}
+			}
+		}
+	};
+
 	// =============================================================================
 	// EFFECTS
 	// =============================================================================
 	useEffect(() => {
-		if (onChange) {
-			const subscription = watch((value) => {
+		const subscription = watch((value) => {
+			if (onChange) {
 				onChange(value, checkIsFormValid());
-			});
+			}
 
-			return () => subscription.unsubscribe();
-		}
-	}, [checkIsFormValid, onChange, watch]);
+			for (const [key, value] of Object.entries(errors)) {
```

Comment:
this may be a problem in larger forms with huge list of errors, as a quick fix, you can subscribe only if there is `onChange` or the errors contain api errors. be sure to do `getFieldState` only for api errors instead of every error